marshmallow/schema.py

```diff
@@ -525,6 +525,16 @@ def dump(self, obj, many=None, update_fields=True, **kwargs):
             error_handler = self.handle_error or self.__error_handler__
             if callable(error_handler):
                 error_handler(errors, obj)
+            exc = ValidationError(
+                errors,
+                field_names=self._marshal.error_field_names,
+                fields=self._marshal.error_fields,
+                data=result,
+                **self._marshal.error_kwargs
+            )
+            self.handle_error(exc, result)
```

Comment:
Here error handler is called second time (see above). Also, according to documentation it is a common practice to raise a different kind of error in error handler. So, by calling `handle_error` you're already violating strictness setting. I guess, you do not need to call `handle_error` at all if not strict.